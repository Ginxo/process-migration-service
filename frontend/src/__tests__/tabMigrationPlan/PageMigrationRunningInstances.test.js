import renderer from "react-test-renderer";
import React from "react";
import PageMigrationRunningInstances
  from "../../component/tabMigrationPlan/wizardExecuteMigration/PageMigrationRunningInstances";
import running_instances from "../../../mock_data/running_instances.json";

test("PageMigrationRunningInstances renders correctly using snapshot", () => {
  const myMock = jest.fn();
  const getRunningInstancesFn = jest.fn(() => {
    return new Promise(function (resolve) {
      resolve({
        data: running_instances,
        headers: [{ "x-total-count": 31 }]
      })
    })
  });
  const tree = renderer
    .create(
      <PageMigrationRunningInstances
        getRunningInstances={getRunningInstancesFn}
        setRunningInstancesIds={myMock}
        onIsValid={myMock}
        migrateAll={false}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(getRunningInstancesFn).toBeCalledTimes(1);
});
