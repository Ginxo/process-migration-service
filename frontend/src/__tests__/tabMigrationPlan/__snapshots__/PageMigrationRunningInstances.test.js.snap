// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMigrationRunningInstances renders correctly using snapshot 1`] = `
<div
  className="container-fluid"
>
  <table
    className="table dataTable table-striped table-bordered table-hover"
    role="grid"
  >
    <thead>
      <tr>
        <th
          aria-label="ID"
          className="sorting_asc sort sort-asc"
          colSpan={1}
          index={0}
          onClick={[Function]}
          rowSpan={1}
        >
          ID
        </th>
        <th
          aria-label="Name"
          className="sorting sort sort-none"
          colSpan={1}
          index={1}
          onClick={[Function]}
          rowSpan={1}
        >
          Name
        </th>
        <th
          aria-label="Description"
          className="sorting sort sort-none"
          colSpan={1}
          index={2}
          onClick={[Function]}
          rowSpan={1}
        >
          Description
        </th>
        <th
          aria-label="Start Time"
          className="sorting sort sort-none"
          colSpan={1}
          index={3}
          onClick={[Function]}
          rowSpan={1}
        >
          Start Time
        </th>
        <th
          aria-label="State"
          className="sorting sort sort-none"
          colSpan={1}
          index={4}
          onClick={[Function]}
          rowSpan={1}
        >
          State
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <form
    className="content-view-pf-pagination table-view-pf-pagination clearfix"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <div
        className="dropup pagination-pf-pagesize btn-group"
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle btn btn-default"
          disabled={false}
          id="pagination-row-dropdown"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          type="button"
        >
          6
           
          <span
            className="caret"
          />
        </button>
        <ul
          aria-labelledby="pagination-row-dropdown"
          className="dropdown-menu"
          role="menu"
        >
          <li
            className="active"
            role="presentation"
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1"
            >
              6
            </a>
          </li>
          <li
            className=""
            role="presentation"
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1"
            >
              10
            </a>
          </li>
          <li
            className=""
            role="presentation"
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1"
            >
              15
            </a>
          </li>
        </ul>
      </div>
      <span>
        per page
      </span>
    </div>
    <div
      className="form-group"
    >
      <span>
        <span
          className="pagination-pf-items-current"
        >
          1
          -
          6
        </span>
         
        of
         
        <span
          className="pagination-pf-items-total"
        />
      </span>
      <ul
        className="pagination pagination-pf-back"
      >
        <li
          className="disabled"
        >
          <a
            href="#"
            onClick={[Function]}
            title="First Page"
          >
            <span
              aria-hidden={true}
              className="fa fa-angle-double-left i"
            />
          </a>
        </li>
        <li
          className="disabled"
        >
          <a
            href="#"
            onClick={[Function]}
            title="Previous Page"
          >
            <span
              aria-hidden={true}
              className="fa fa-angle-left i"
            />
          </a>
        </li>
      </ul>
      <label
        className="sr-only control-label"
      >
        Current Page
      </label>
      <input
        className="pagination-pf-page form-control"
        onChange={[Function]}
        type="text"
        value={1}
      />
      <span>
         
        of
         
        <span
          className="pagination-pf-pages"
        >
          NaN
        </span>
      </span>
      <ul
        className="pagination pagination-pf-forward"
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
            title="Next Page"
          >
            <span
              aria-hidden={true}
              className="fa fa-angle-right i"
            />
          </a>
        </li>
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
            title="Last Page"
          >
            <span
              aria-hidden={true}
              className="fa fa-angle-double-right i"
            />
          </a>
        </li>
      </ul>
    </div>
  </form>
</div>
`;
